incubator/command-config/src/main/java/io/aklivity/zilla/runtime/command/config/internal/openapi/http/proxy/OpenApiHttpProxyConfigGenerator.java

```diff
@@ -0,0 +1,505 @@
+
+    private BindingConfigBuilder<NamespaceConfigBuilder<NamespaceConfig>> injectTlsTcpRoute(
+        BindingConfigBuilder<NamespaceConfigBuilder<NamespaceConfig>> binding)
+    {
+        if (isTlsEnabled && !isPlainEnabled)
+        {
+            binding
+                .route()
+                    .exit("tls_server0")
+                    .build();
+        }
+        return binding;
+    }
+
+    private BindingConfigBuilder<NamespaceConfigBuilder<NamespaceConfig>> injectPlainAndTlsTcpRoutes(
+        BindingConfigBuilder<NamespaceConfigBuilder<NamespaceConfig>> binding)
+    {
+        if (isPlainEnabled && isTlsEnabled)
+        {
+            binding
+                .route()
+                    .when(TcpConditionConfig::builder)
+                        .ports(httpPorts)
+                        .build()
+                    .exit("http_server0")
+                    .build()
+                .route()
+                    .when(TcpConditionConfig::builder)
+                        .ports(httpsPorts)
+                        .build()
+                    .exit("tls_server0")
+                    .build();
+        }
+        return binding;
+    }
+
+    private NamespaceConfigBuilder<NamespaceConfig> injectTlsServer(
+        NamespaceConfigBuilder<NamespaceConfig> namespace)
+    {
+        if (isTlsEnabled)
+        {
+            namespace
+                .binding()
+                    .name("tls_server0")
+                    .type("tls")
+                    .kind(SERVER)
+                    .options(TlsOptionsConfig::builder)
+                        .keys(List.of("")) // env
+                        .sni(List.of("")) // env
+                        .alpn(List.of("")) // env
+                        .build()
+                    .vault("server")
+                    .route()
+                        .exit("http_server0")
+                        .build()
+                    .build();
+        }
+        return namespace;
+    }
+
+    private HttpOptionsConfigBuilder<BindingConfigBuilder<NamespaceConfigBuilder<NamespaceConfig>>> injectHttpServerOptions(
+        HttpOptionsConfigBuilder<BindingConfigBuilder<NamespaceConfigBuilder<NamespaceConfig>>> options)
+    {
+        if (isJwtEnabled)
+        {
+            options
+                .authorization()
+                    .name("jwt0")
+                    .credentials()
+                        .header()
+                            .name("authorization")
+                            .pattern("Bearer {credentials}")
+                            .build()
+                    .build()
+                .build();
+        }
+        return options;
+    }
+
+    private BindingConfigBuilder<NamespaceConfigBuilder<NamespaceConfig>> injectHttpServerRoutes(
+        BindingConfigBuilder<NamespaceConfigBuilder<NamespaceConfig>> binding)
+    {
+        for (String path : openApi.paths.keySet())
+        {
+            PathItem2 item = PathItem2.of(openApi.paths.get(path));
```

Comment:
If we are going to use `model2` classes, then perhaps it would be helpful to massage their representation to better suit iteration for generation of zilla.yaml.

For example, perhaps the concept of operation:
```
Operation
{
    String operationId
    String method
    String path
    List<String> roles
}
```

Reply:
I checked this approach but didn't seem to work out.

- Using `Operation2` with `method` and `path` fields didn't result in any simplification of the code in `injectHttpServerRoutes`. We would need to create an extra object to contain and read values that we already have anyway, so this change did not seem to have a point.
- Moving the complex logic from `injectHttpServerRouteGuarded` to `Operation2` could be nice, however that code needs to access the instance variable `securitySchemes`, so we'd need to pass it when creating the `Operation2` object which looked a bit contrived to me.

I've finally undone all my changes, I'd prefer to keep it as is (at least for now); pls LMK if you insist.

Reply:
Perhaps an inner class of the generator instead of a separate model2 package?

Reply:
I wouldn't move the models to inner classes as we will have multiple generators (for asyncapi anyway, but potentially for openapi as well), so it makes sense that various generators can reuse the same model classes.